src/osx/textctrl_osx.cpp

```diff
@@ -681,16 +681,74 @@ void wxTextWidgetImpl::Redo()
 {
 }
 
-long wxTextWidgetImpl::XYToPosition(long WXUNUSED(x), long WXUNUSED(y)) const
+long wxTextWidgetImpl::XYToPosition(long x, long y) const
 {
-    return 0 ;
-}
+    const wxString& content = GetStringValue() ;
+    long line = 1, cur_pos = 0;
 
-bool wxTextWidgetImpl::PositionToXY(long WXUNUSED(pos),
-                                    long *WXUNUSED(x),
-                                    long *WXUNUSED(y)) const
-{
-    return false ;
+    for ( wxString::const_iterator it = content.begin(); it != content.end(); ++it )
+    {
+        if (line == y)
+        {
+            long col = 1;
+            for ( ; it != content.end(); ++it, ++col)
+            {
+                if (col == x)
+                {
+                    break;
+                }
+                else if (*it == '/n') 
+                {
+                    cur_pos = wxNOT_FOUND;
+                }
+                cur_pos++;
+            }
+            break;
+        }
+        cur_pos++;
+        if (*it == '\n')
+        {
+            line++;
+        }
+    }
+    return line == y ? cur_pos : wxNOT_FOUND ;
+}
+
+bool wxTextWidgetImpl::PositionToXY(long pos,
+                                    long *x,
+                                    long *y) const
+{
+    const wxString& content = GetStringValue() ;
+    long line = 1, line_length = 0, cur_pos = 0;
+    
+    // Set the column & line numbers to 0 by default 
+    // so that the developer gets a sane/easy to check result
+    *x = *y = 0;
+    
+    // Return wxNOT_FOUND if the requested position is 
+    // larger than the content size or less than zero.
+    if (pos > content.Length() || pos < 0) 
```

Comment:
As I said, I think it's not worth it to optimize for what is fundamentally an invalid input value. We do need to handle them correctly, but having special checks to handle them fast seems unnecessary. It's not a big deal though, it's just that you'd also need a cast here to compare signed `pos` with unsigned `length()` if you keep it (because you ought to use the standard string-like `length()` and not the old wx1.x-compatible `Length()), so it's just a waste of time and code lines for nothing...